rust/agents/relayer/src/msg/metadata/multisig/base.rs

```diff
@@ -12,24 +12,27 @@ use hyperlane_core::{Checkpoint, HyperlaneMessage, SignatureWithSigner, H256};
 use strum::Display;
 use tracing::{debug, info};
 
+use crate::error::RelayerError;
 use crate::msg::metadata::BaseMetadataBuilder;
 use crate::msg::metadata::MetadataBuilder;
 
 #[derive(new)]
 pub struct MultisigMetadata {
     checkpoint: Checkpoint,
     signatures: Vec<SignatureWithSigner>,
+    merkle_leaf_id: Option<u32>,
```

Comment:
nit: similarly this should clarify this is the index, e.g. `merkle_leaf_index` or `message_leaf_index` or something